src/effects/VST/VSTEffect.cpp

```diff
@@ -1087,11 +1086,11 @@ bool VSTEffect::RealtimeAddProcessor(
    if (clen == 0) {
       callDispatcher(effBeginSetProgram, 0, 0, NULL, 0.0);
       for (int i = 0; i < mAEffect->numParams; i++)
-         slave->callSetParameter(i, callGetParameter(i));
+         slave->callSetParameterB(i, callGetParameter(i));
```

Comment:
I did not mention this before, but if you look for the analogous uses of `StoreSettings` on AudioUnit instances, one of them happens here.  In step 4 you might make a similar change here too, eliminating the loop over parameters, and the beginnin and end of set-program, which would happen inside `StoreSettings`.

Reply:
OK done that in commit 4.2 (yet to be pushed at this moment) - you will see that to make it work I also had to add a call to `FetchSettings` inside `Automate`.